src/Rule/AbstractCharCase.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ *
+ * This file is part of Aura for PHP.
+ *
+ * @license http://opensource.org/licenses/bsd-license.php BSD
+ *
+ */
+namespace Aura\Filter\Rule;
+
+use Aura\Filter\Exception;
+
+/**
+ *
+ * Abstract rule for character case filters; supports the `mbstring`
+ * extension.
+ *
+ * @package Aura.Filter
+ *
+ */
+abstract class AbstractCharCase extends AbstractStrlen
```

Comment:
may be you missed to add this class? I am looking where the `$this->substr` is called over https://github.com/auraphp/Aura.Filter/pull/106/files#diff-b24d614b81004a326954313c020d1a83R93